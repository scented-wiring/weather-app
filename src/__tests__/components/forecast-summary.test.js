import React from "react";
import { render } from "@testing-library/react";
import ForecastSummary from "../../components/forecast-summary";

describe("ForecastSummary", () => {
  it("renders correctly", () => {
    const { asFragment } = render(
      <ForecastSummary
        date={1525046400000}
        temperature={13}
        description="mockDescription"
        icon="800"
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the correct props", () => {
    const { getByTestId } = render(
      <ForecastSummary
        date={1525046400000}
        temperature={13}
        description="mockDescription"
        icon="200"
      />
    );

    expect(getByTestId("date-id")).toBeTruthy();
    expect(getByTestId("icon-id")).toBeTruthy();
    expect(getByTestId("temperature-id")).toBeTruthy();
    expect(getByTestId("description-id")).toBeTruthy();
  });
});
