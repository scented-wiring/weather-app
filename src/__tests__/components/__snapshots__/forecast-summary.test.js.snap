// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummary renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-summary"
  >
    <div
      class="date"
      data-testid="date-id"
    >
      Mon 30th Apr
    </div>
    <div
      class="temperature"
      data-testid="temperature-id"
    >
      13Â°c
    </div>
    <div
      class="description"
      data-testid="description-id"
    >
      mockDescription
    </div>
    <div
      class="icon"
      data-testid="icon-id"
    >
      <i
        class="wi wi-day-sunny"
        name="owm"
      />
    </div>
    <button>
      More Details
    </button>
  </div>
</DocumentFragment>
`;
